AxisRotationConstraint.cs

```diff
@@ -0,0 +1,138 @@
+using Microsoft.MixedReality.Toolkit;
+using Microsoft.MixedReality.Toolkit.SpatialManipulation;
+using UnityEngine;
+
+public class AxisRotationConstraint : TransformConstraint

```

Comment:
Both `AxisMovementConstraint` and `AxisRotationConstraint` share alot of the same code and structure.   Can you please refactor to reduce code duplication.